In [1]:
library('tidyverse')

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors


# 1 Data analysis

1. Import the data set. Make sure the column names are imported as intended.

In [2]:
data <- read_csv('../../../../data/wage2015_subsample_inference.csv')
data

Rows: 5150 Columns: 21
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
dbl (21): rownames, wage, lwage, sex, shs, hsg, scl, clg, ad, mw, so, we, ne...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


rownames wage      lwage    sex shs hsg scl clg ad mw ⋯ we ne exp1 exp2   
1     10       9.615385 2.263364 1   0   0   0   1   0  0  ⋯ 0  1   7.0  0.4900
2     12      48.076923 3.872802 0   0   0   0   1   0  0  ⋯ 0  1  31.0  9.6100
3     15      11.057692 2.403126 0   0   1   0   0   0  0  ⋯ 0  1  18.0  3.2400
4     18      13.942308 2.634928 1   0   0   0   0   1  0  ⋯ 0  1  25.0  6.2500
5     19      28.846154 3.361977 1   0   0   0   1   0  0  ⋯ 0  1  22.0  4.8400
6     30      11.730769 2.462215 1   0   0   0   1   0  0  ⋯ 0  1   1.0  0.0100
7     43      19.230769 2.956512 1   0   1   0   0   0  0  ⋯ 0  1  42.0 17.6400
8     44      19.230769 2.956512 0   0   1   0   0   0  0  ⋯ 0  1  37.0 13.6900
9     47      12.000000 2.484907 1   0   1   0   0   0  0  ⋯ 0  1  31.0  9.6100
10    71      19.230769 2.956512 1   0   0   0   1   0  0  ⋯ 0  1   4.0  0.1600
11    73      17.307692 2.851151 1   0   1   0   0   0  0  ⋯ 0  1   7.0  0.4900
12    77      12.019231 2.486508 0   0   1   0   0   0  0  ⋯ 0  1  30.0  9.0000
13    84      12.019231 2.486508 1   0   0   1   0   0  0  ⋯ 0  1   5.5  0.3025
14    89      13.461538 2.599837 1   0   0   1   0   0  0  ⋯ 0  1  20.5  4.2025
15    96      16.346154 2.793993 1   0   0   0   1   0  0  ⋯ 0  1  25.0  6.2500
16   113      27.884615 3.328075 0   0   1   0   0   0  0  ⋯ 0  1  16.0  2.5600
17   115      21.600000 3.072693 0   0   1   0   0   0  0  ⋯ 0  1  27.0  7.2900
18   119       8.653846 2.158004 0   0   0   1   0   0  0  ⋯ 0  1   3.5  0.1225
19   128      19.230769 2.956512 0   0   1   0   0   0  0  ⋯ 0  1   6.0  0.3600
20   129      13.186813 2.579217 0   0   1   0   0   0  0  ⋯ 0  1   8.0  0.6400
21   135      10.683761 2.368725 1   0   0   1   0   0  0  ⋯ 0  1   6.5  0.4225
22   144      11.538462 2.445686 1   0   0   1   0   0  0  ⋯ 0  1   8.5  0.7225
23   145      17.788462 2.878550 1   0   0   0   0   1  0  ⋯ 0  1  11.0  1.2100
24   149      19.230769 2.956512 0   0   0   0   1   0  0  ⋯ 0  1  11.0  1.2100
25   152      16.304348 2.791432 1   0   0   1   0   0  0  ⋯ 0  1  24.5  6.0025
26   157      19.230769 2.956512 0   0   1   0   0   0  0  ⋯ 0  1   8.0  0.6400
27   158      14.423077 2.668829 1   0   1   0   0   0  0  ⋯ 0  1   5.0  0.2500
28   160      12.000000 2.484907 0   0   0   1   0   0  0  ⋯ 0  1   4.5  0.2025
29   164      16.826923 2.822980 0   0   0   0   1   0  0  ⋯ 0  1   8.0  0.6400
30   176      19.670330 2.979111 0   0   0   0   1   0  0  ⋯ 0  1  26.0  6.7600
⋮    ⋮        ⋮         ⋮        ⋮   ⋮   ⋮   ⋮   ⋮   ⋮  ⋮  ⋱ ⋮  ⋮  ⋮    ⋮      
5121 32537    13.986014 2.638058 1   0   0   0   0   1  0  ⋯ 1  0   7.0  0.4900
5122 32538    15.865385 2.764140 0   0   0   0   0   1  0  ⋯ 1  0   2.0  0.0400
5123 32539    38.461538 3.649659 1   0   0   1   0   0  0  ⋯ 1  0   7.5  0.5625
5124 32540    28.846154 3.361977 0   0   0   0   1   0  0  ⋯ 1  0   5.0  0.2500
5125 32548    24.475524 3.197674 1   0   0   0   0   1  0  ⋯ 1  0  24.0  5.7600
5126 32551    27.884615 3.328075 0   0   0   1   0   0  0  ⋯ 1  0  28.5  8.1225
5127 32559     8.653846 2.158004 1   0   1   0   0   0  0  ⋯ 1  0  25.0  6.2500
5128 32560    12.019231 2.486508 0   0   1   0   0   0  0  ⋯ 1  0  19.0  3.6100
5129 32561    38.461538 3.649659 0   0   0   0   1   0  0  ⋯ 1  0   8.0  0.6400
5130 32564    12.500000 2.525729 1   0   1   0   0   0  0  ⋯ 1  0  12.0  1.4400
5131 32565    35.256410 3.562647 0   0   0   0   0   1  0  ⋯ 1  0   6.0  0.3600
5132 32567    48.076923 3.872802 1   0   0   0   1   0  0  ⋯ 1  0  25.0  6.2500
5133 32571     9.615385 2.263364 0   0   1   0   0   0  0  ⋯ 1  0  20.0  4.0000
5134 32572    12.019231 2.486508 0   0   1   0   0   0  0  ⋯ 1  0  17.0  2.8900
5135 32579    12.019231 2.486508 0   0   0   1   0   0  0  ⋯ 1  0  31.5  9.9225
5136 32580    12.980769 2.563469 0   0   1   0   0   0  0  ⋯ 1  0  15.0  2.2500
5137 32581    26.442308 3.274965 1   0   0   0   1   0  0  ⋯ 1  0  34.0 11.5600
5138 32590    13.461538 2.599837 1   0   1   0   0   0  0  ⋯ 1  0   8.0  0.6400
5139 

In [3]:
data <- select(data, -rownames)
data

wage      lwage    sex shs hsg scl clg ad mw so we ne exp1 exp2   
1     9.615385 2.263364 1   0   0   0   1   0  0  0  0  1   7.0  0.4900
2    48.076923 3.872802 0   0   0   0   1   0  0  0  0  1  31.0  9.6100
3    11.057692 2.403126 0   0   1   0   0   0  0  0  0  1  18.0  3.2400
4    13.942308 2.634928 1   0   0   0   0   1  0  0  0  1  25.0  6.2500
5    28.846154 3.361977 1   0   0   0   1   0  0  0  0  1  22.0  4.8400
6    11.730769 2.462215 1   0   0   0   1   0  0  0  0  1   1.0  0.0100
7    19.230769 2.956512 1   0   1   0   0   0  0  0  0  1  42.0 17.6400
8    19.230769 2.956512 0   0   1   0   0   0  0  0  0  1  37.0 13.6900
9    12.000000 2.484907 1   0   1   0   0   0  0  0  0  1  31.0  9.6100
10   19.230769 2.956512 1   0   0   0   1   0  0  0  0  1   4.0  0.1600
11   17.307692 2.851151 1   0   1   0   0   0  0  0  0  1   7.0  0.4900
12   12.019231 2.486508 0   0   1   0   0   0  0  0  0  1  30.0  9.0000
13   12.019231 2.486508 1   0   0   1   0   0  0  0  0  1   5.5  0.3025
14   13.461538 2.599837 1   0   0   1   0   0  0  0  0  1  20.5  4.2025
15   16.346154 2.793993 1   0   0   0   1   0  0  0  0  1  25.0  6.2500
16   27.884615 3.328075 0   0   1   0   0   0  0  0  0  1  16.0  2.5600
17   21.600000 3.072693 0   0   1   0   0   0  0  0  0  1  27.0  7.2900
18    8.653846 2.158004 0   0   0   1   0   0  0  0  0  1   3.5  0.1225
19   19.230769 2.956512 0   0   1   0   0   0  0  0  0  1   6.0  0.3600
20   13.186813 2.579217 0   0   1   0   0   0  0  0  0  1   8.0  0.6400
21   10.683761 2.368725 1   0   0   1   0   0  0  0  0  1   6.5  0.4225
22   11.538462 2.445686 1   0   0   1   0   0  0  0  0  1   8.5  0.7225
23   17.788462 2.878550 1   0   0   0   0   1  0  0  0  1  11.0  1.2100
24   19.230769 2.956512 0   0   0   0   1   0  0  0  0  1  11.0  1.2100
25   16.304348 2.791432 1   0   0   1   0   0  0  0  0  1  24.5  6.0025
26   19.230769 2.956512 0   0   1   0   0   0  0  0  0  1   8.0  0.6400
27   14.423077 2.668829 1   0   1   0   0   0  0  0  0  1   5.0  0.2500
28   12.000000 2.484907 0   0   0   1   0   0  0  0  0  1   4.5  0.2025
29   16.826923 2.822980 0   0   0   0   1   0  0  0  0  1   8.0  0.6400
30   19.670330 2.979111 0   0   0   0   1   0  0  0  0  1  26.0  6.7600
⋮    ⋮         ⋮        ⋮   ⋮   ⋮   ⋮   ⋮   ⋮  ⋮  ⋮  ⋮  ⋮  ⋮    ⋮      
5121 13.986014 2.638058 1   0   0   0   0   1  0  0  1  0   7.0  0.4900
5122 15.865385 2.764140 0   0   0   0   0   1  0  0  1  0   2.0  0.0400
5123 38.461538 3.649659 1   0   0   1   0   0  0  0  1  0   7.5  0.5625
5124 28.846154 3.361977 0   0   0   0   1   0  0  0  1  0   5.0  0.2500
5125 24.475524 3.197674 1   0   0   0   0   1  0  0  1  0  24.0  5.7600
5126 27.884615 3.328075 0   0   0   1   0   0  0  0  1  0  28.5  8.1225
5127  8.653846 2.158004 1   0   1   0   0   0  0  0  1  0  25.0  6.2500
5128 12.019231 2.486508 0   0   1   0   0   0  0  0  1  0  19.0  3.6100
5129 38.461538 3.649659 0   0   0   0   1   0  0  0  1  0   8.0  0.6400
5130 12.500000 2.525729 1   0   1   0   0   0  0  0  1  0  12.0  1.4400
5131 35.256410 3.562647 0   0   0   0   0   1  0  0  1  0   6.0  0.3600
5132 48.076923 3.872802 1   0   0   0   1   0  0  0  1  0  25.0  6.2500
5133  9.615385 2.263364 0   0   1   0   0   0  0  0  1  0  20.0  4.0000
5134 12.019231 2.486508 0   0   1   0   0   0  0  0  1  0  17.0  2.8900
5135 12.019231 2.486508 0   0   0   1   0   0  0  0  1  0  31.5  9.9225
5136 12.980769 2.563469 0   0   1   0   0   0  0  0  1  0  15.0  2.2500
5137 26.442308 3.274965 1   0   0   0   1   0  0  0  1  0  34.0 11.5600
5138 13.461538 2.599837 1   0   1   0   0   0  0  0  1  0   8.0  0.6400
5139 19.711538 2.981204 1   0   0   0   0   1  0  0  1  0   8.0  0.6400
5140 21.153846 3.051822 0   0   0   0   1   0  0  0  1  0  28.0  7.8400
5141 45.546559 3.818735 1   0   0   0   1   0  0  0  1  0   5.0  0.2500
5142 22.596154 3.117780 0   0   1   0   0   0  0  0  1  0  15.0  2.2500
5143 16.826923 2.822980 0   0   1   0   0   0  0  0  1  0  11.0  1.2100
5144 24.038462 3.179655 1   0   0   0   1   0

2. Are there missing values? Display the number of missings (if any) of each variable in the data set.

In [4]:
data %>% summarise_all(~sum(is.na(.)))

wage lwage sex shs hsg scl clg ad mw so we ne exp1 exp2 exp3 exp4 occ occ2
1 0    0     0   0   0   0   0   0  0  0  0  0  0    0    0    0    0   0   
  ind ind2
1 0   0

3. Report descriptive statistics of the variables (mean, standard deviation, percetiles, etc.). Interpret your results.

In [5]:
summary(data)

      wage             lwage            sex              shs        
 Min.   :  3.022   Min.   :1.106   Min.   :0.0000   Min.   :0.0000  
 1st Qu.: 13.461   1st Qu.:2.600   1st Qu.:0.0000   1st Qu.:0.0000  
 Median : 19.231   Median :2.957   Median :0.0000   Median :0.0000  
 Mean   : 23.410   Mean   :2.971   Mean   :0.4445   Mean   :0.0233  
 3rd Qu.: 27.778   3rd Qu.:3.324   3rd Qu.:1.0000   3rd Qu.:0.0000  
 Max.   :528.846   Max.   :6.271   Max.   :1.0000   Max.   :1.0000  
      hsg              scl              clg               ad        
 Min.   :0.0000   Min.   :0.0000   Min.   :0.0000   Min.   :0.0000  
 1st Qu.:0.0000   1st Qu.:0.0000   1st Qu.:0.0000   1st Qu.:0.0000  
 Median :0.0000   Median :0.0000   Median :0.0000   Median :0.0000  
 Mean   :0.2439   Mean   :0.2781   Mean   :0.3177   Mean   :0.1371  
 3rd Qu.:0.0000   3rd Qu.:1.0000   3rd Qu.:1.0000   3rd Qu.:0.0000  
 Max.   :1.0000   Max.   :1.0000   Max.   :1.0000   Max.   :1.0000  
       mw               so        

As most of the variables are dummy variables, only wage is interprated based in these descriptive statistics. The average wage in the sample is 23.410 dollars per hour. The 25% richest earn 27.78 dollars per hour or more, while the 25% poorest earn 13.45 or less. The median earning per hour is 19.23 dollars, which is lower than the average wage and shows the asymmetry in the wage distribution of the sample. 

4. How many women with a college graduate degree (clg) or above have a wage corresponding to the 25% richest of the sample? Report the dataframe corresponding to this conditions and the result.

In [6]:
data_women_clg_75 <- filter(data, (sex == 1) & (clg==1|ad==1) & (wage > quantile(data$wage,probs = c(0.25,0.5,0.75))[3]))
data_women_clg_75

wage      lwage    sex shs hsg scl clg ad mw so we ne exp1 exp2  exp3  
1    28.84615 3.361977 1   0   0   0   1   0  0  0  0  1  22    4.84 10.648
2    42.30769 3.744969 1   0   0   0   0   1  0  0  0  1   0    0.00  0.000
3    41.20879 3.718652 1   0   0   0   1   0  0  0  0  1  11    1.21  1.331
4   100.00000 4.605170 1   0   0   0   0   1  0  0  0  1  27    7.29 19.683
5    33.65385 3.516127 1   0   0   0   1   0  0  0  0  1   4    0.16  0.064
6    36.05769 3.585120 1   0   0   0   1   0  0  0  0  1  23    5.29 12.167
7    31.25000 3.442019 1   0   0   0   1   0  0  0  0  1  15    2.25  3.375
8    28.84615 3.361977 1   0   0   0   1   0  0  0  0  1   9    0.81  0.729
9    48.07692 3.872802 1   0   0   0   1   0  0  0  0  1  24    5.76 13.824
10   34.61538 3.544298 1   0   0   0   1   0  0  0  0  1  18    3.24  5.832
11   31.25000 3.442019 1   0   0   0   0   1  0  0  0  1  29    8.41 24.389
12   28.57143 3.352407 1   0   0   0   1   0  0  0  0  1  13    1.69  2.197
13   28.50000 3.349904 1   0   0   0   1   0  0  0  0  1  19    3.61  6.859
14   36.05769 3.585120 1   0   0   0   0   1  0  0  0  1  38   14.44 54.872
15   31.25000 3.442019 1   0   0   0   1   0  0  0  0  1  18    3.24  5.832
16   38.46154 3.649659 1   0   0   0   1   0  0  0  0  1  19    3.61  6.859
17   40.48583 3.700952 1   0   0   0   0   1  0  0  0  1  33   10.89 35.937
18   32.05128 3.467337 1   0   0   0   1   0  0  0  0  1   4    0.16  0.064
19   28.84615 3.361977 1   0   0   0   0   1  0  0  0  1  12    1.44  1.728
20   43.01619 3.761577 1   0   0   0   1   0  0  0  0  1   1    0.01  0.001
21   62.50000 4.135167 1   0   0   0   0   1  0  0  0  1   6    0.36  0.216
22   34.18803 3.531876 1   0   0   0   1   0  0  0  0  1  17    2.89  4.913
23   48.07692 3.872802 1   0   0   0   1   0  0  0  0  1  25    6.25 15.625
24   29.97692 3.400428 1   0   0   0   0   1  0  0  0  1   0    0.00  0.000
25   31.25000 3.442019 1   0   0   0   1   0  0  0  0  1  27    7.29 19.683
26   48.58300 3.883274 1   0   0   0   1   0  0  0  0  1  28    7.84 21.952
27   30.76923 3.426515 1   0   0   0   1   0  0  0  0  1   3    0.09  0.027
28   50.84685 3.928818 1   0   0   0   1   0  0  0  0  1  38   14.44 54.872
29   28.84615 3.361977 1   0   0   0   0   1  0  0  0  1   3    0.09  0.027
30   28.84615 3.361977 1   0   0   0   1   0  0  0  0  1   3    0.09  0.027
⋮   ⋮         ⋮        ⋮   ⋮   ⋮   ⋮   ⋮   ⋮  ⋮  ⋮  ⋮  ⋮  ⋮    ⋮     ⋮     
383 36.05769  3.585120 1   0   0   0   1   0  0  0  1  0   1    0.01  0.001
384 34.93590  3.553515 1   0   0   0   0   1  0  0  1  0  31    9.61 29.791
385 36.05769  3.585120 1   0   0   0   1   0  0  0  1  0  30    9.00 27.000
386 47.11538  3.852600 1   0   0   0   0   1  0  0  1  0  16    2.56  4.096
387 38.00000  3.637586 1   0   0   0   0   1  0  0  1  0  16    2.56  4.096
388 41.52098  3.726199 1   0   0   0   0   1  0  0  1  0  36   12.96 46.656
389 39.90385  3.686473 1   0   0   0   1   0  0  0  1  0   9    0.81  0.729
390 31.25096  3.442050 1   0   0   0   1   0  0  0  1  0   3    0.09  0.027
391 36.53846  3.598365 1   0   0   0   1   0  0  0  1  0  17    2.89  4.913
392 28.84615  3.361977 1   0   0   0   1   0  0  0  1  0   6    0.36  0.216
393 34.61538  3.544298 1   0   0   0   0   1  0  0  1  0   1    0.01  0.001
394 53.84615  3.986131 1   0   0   0   0   1  0  0  1  0  31    9.61 29.791
395 44.87179  3.803809 1   0   0   0   1   0  0  0  1  0  38   14.44 54.872
396 33.65385  3.516127 1   0   0   0   0   1  0  0  1  0   6    0.36  0.216
397 40.86538  3.710283 1   0   0   0   1   0  0  0  1  0   9    0.81  0.729
398 67.30769  4.209275 1   0   0   0   0   1  0  0  1  0   7    0.49  0.343
399 50.48077  3.921592 1   0   0   0   0   1  0  0  1  0  17    2.89  4.913
400 32.69231  3.487140 1   0   0   0   0   1  0  0  1  0   8    0.64  0.512
401 48.07692  3.872802 1   0   0   0   0   1  0  0  1  0  19    3.61  6.859
402 33.65385  3.516127 1   0   0   0   0   1  0  0  1  0  12    1.44  1.728
403 40.38462  3.698449 1   0   0   0   1   0  0  0  

In [7]:
print(paste(dim(data_women_clg_75)[1],'women with a college graduate degree (clg) or above have a wage corresponding to the 25% richest of the sample'))

[1] "412 women with a college graduate degree (clg) or above have a wage corresponding to the 25% richest of the sample"


5. How many men with a high school graduate degree (hsg) or below have a wage corresponding to the 25% richest of the sample? Report the dataframe corresponding to this conditions and the result.

In [8]:
data_men_hsg_75 <- filter(data, (sex == 0) & (shs==1|hsg==1) & (wage > quantile(data$wage,probs = c(0.25,0.5,0.75))[3]))
data_men_hsg_75

wage      lwage    sex shs hsg scl clg ad mw so we ne exp1 exp2    exp3    
1   27.88462  3.328075 0   0   1   0   0   0  0  0  0  1  16    2.56    4.096  
2   28.84615  3.361977 0   0   1   0   0   0  0  0  0  1  41   16.81   68.921  
3   28.84615  3.361977 0   0   1   0   0   0  0  0  0  1  33   10.89   35.937  
4   28.84615  3.361977 0   0   1   0   0   0  0  0  0  1  40   16.00   64.000  
5   29.71429  3.391628 0   0   1   0   0   0  0  0  0  1   8    0.64    0.512  
6   35.57692  3.571697 0   0   1   0   0   0  0  0  0  1   8    0.64    0.512  
7   36.32479  3.592500 0   0   1   0   0   0  0  0  0  1  30    9.00   27.000  
8   43.26923  3.767442 0   0   1   0   0   0  0  0  0  1  29    8.41   24.389  
9   43.26923  3.767442 0   0   1   0   0   0  0  0  0  1  29    8.41   24.389  
10  43.26923  3.767442 0   0   1   0   0   0  0  0  0  1  29    8.41   24.389  
11  43.26923  3.767442 0   0   1   0   0   0  0  0  0  1  29    8.41   24.389  
12  28.84615  3.361977 0   0   1   0   0   0  0  0  0  1   8    0.64    0.512  
13  28.07692  3.334948 0   0   1   0   0   0  0  0  0  1  22    4.84   10.648  
14  45.67308  3.821509 0   0   1   0   0   0  0  0  0  1  16    2.56    4.096  
15  37.95547  3.636414 0   0   1   0   0   0  0  0  0  1  32   10.24   32.768  
16  31.37255  3.445933 0   0   1   0   0   0  0  0  0  1  15    2.25    3.375  
17  33.65385  3.516127 0   0   1   0   0   0  0  0  0  1  28    7.84   21.952  
18  28.84567  3.361960 0   1   0   0   0   0  0  0  0  1  42   17.64   74.088  
19  33.65385  3.516127 0   0   1   0   0   0  0  0  0  1  32   10.24   32.768  
20  50.48077  3.921592 0   0   1   0   0   0  0  0  0  1   6    0.36    0.216  
21  43.26923  3.767442 0   0   1   0   0   0  0  0  0  1  34   11.56   39.304  
22  28.84615  3.361977 0   0   1   0   0   0  0  0  0  1  22    4.84   10.648  
23  28.84615  3.361977 0   0   1   0   0   0  0  0  0  1  10    1.00    1.000  
24  28.84615  3.361977 0   0   1   0   0   0  0  0  0  1   5    0.25    0.125  
25  28.84615  3.361977 0   0   1   0   0   0  0  0  0  1  40   16.00   64.000  
26  32.69231  3.487140 0   0   1   0   0   0  0  0  0  1  32   10.24   32.768  
27  33.41346  3.508959 0   0   1   0   0   0  0  0  0  1  27    7.29   19.683  
28  33.65385  3.516127 0   0   1   0   0   0  0  0  0  1  23    5.29   12.167  
29  33.06827  3.498574 0   0   1   0   0   0  1  0  0  0  13    1.69    2.197  
30  50.48077  3.921592 0   0   1   0   0   0  1  0  0  0  16    2.56    4.096  
⋮   ⋮         ⋮        ⋮   ⋮   ⋮   ⋮   ⋮   ⋮  ⋮  ⋮  ⋮  ⋮  ⋮    ⋮       ⋮       
86   38.46154 3.649659 0   0   1   0   0   0  0  0  1  0  40.0 16.0000 64.00000
87   32.50000 3.481240 0   0   1   0   0   0  0  0  1  0  10.0  1.0000  1.00000
88   29.72028 3.391830 0   0   1   0   0   0  0  0  1  0  25.0  6.2500 15.62500
89   37.50000 3.624341 0   0   1   0   0   0  0  0  1  0  25.0  6.2500 15.62500
90   35.09615 3.558092 0   1   0   0   0   0  0  0  1  0  34.0 11.5600 39.30400
91   28.84615 3.361977 0   1   0   0   0   0  0  0  1  0  13.0  1.6900  2.19700
92   28.84615 3.361977 0   1   0   0   0   0  0  0  1  0  11.0  1.2100  1.33100
93   31.80473 3.459615 0   0   1   0   0   0  0  0  1  0   9.0  0.8100  0.72900
94   31.73077 3.457287 0   0   1   0   0   0  0  0  1  0  16.0  2.5600  4.09600
95   30.00000 3.401197 0   0   1   0   0   0  0  0  1  0  17.0  2.8900  4.91300
96   28.04487 3.333806 0   0   1   0   0   0  0  0  1  0  35.0 12.2500 42.87500
97   38.07692 3.639608 0   0   1   0   0   0  0  0  1  0   8.0  0.6400  0.51200
98  153.84615 5.035953 0   0   1   0   0   0  0  0  1  0   8.0  0.6400  0.51200
99  528.84567 6.270697 0   1   0   0   0   0  0  0  1  0  19.0  3.6100  6.85900
100  34.61538 3.544298 0   0   1   0   0   0  0  0  1  0  18.0  3.2400  5.83200
101  28.57143 3.352407 0   0   1   0   0   0  0  0  1  0  17.0  2.8900  4.91300
102  60.09615 4.095946 0   0   1   0   0   0  0  0  1  0  39.0 15.2100 59.31900
103  34.61538 3.544298 0   0   1   0   0   0  0  0  1  0  12.0  1.4400  1.72800
104 

In [9]:
print(paste(dim(data_men_hsg_75)[1],'men with a high school graduate degree (hsg) or below have a wage corresponding to the 25% richest of the sample'))

[1] "115 men with a high school graduate degree (hsg) or below have a wage corresponding to the 25% richest of the sample"


6. Create two dataframes. One containing only the log(wage) and the other containig every variable of the data set but the wage related variables.

In [10]:
log_wage <- select(data, lwage)
log_wage


lwage   
1    2.263364
2    3.872802
3    2.403126
4    2.634928
5    3.361977
6    2.462215
7    2.956512
8    2.956512
9    2.484907
10   2.956512
11   2.851151
12   2.486508
13   2.486508
14   2.599837
15   2.793993
16   3.328075
17   3.072693
18   2.158004
19   2.956512
20   2.579217
21   2.368725
22   2.445686
23   2.878550
24   2.956512
25   2.791432
26   2.956512
27   2.668829
28   2.484907
29   2.822980
30   2.979111
⋮    ⋮       
5121 2.638058
5122 2.764140
5123 3.649659
5124 3.361977
5125 3.197674
5126 3.328075
5127 2.158004
5128 2.486508
5129 3.649659
5130 2.525729
5131 3.562647
5132 3.872802
5133 2.263364
5134 2.486508
5135 2.486508
5136 2.563469
5137 3.274965
5138 2.599837
5139 2.981204
5140 3.051822
5141 3.818735
5142 3.117780
5143 2.822980
5144 3.179655
5145 2.628007
5146 2.692546
5147 3.138833
5148 3.649659
5149 3.495508
5150 2.851151

In [11]:
features <- select(data, -wage, -lwage)
features

sex shs hsg scl clg ad mw so we ne exp1 exp2    exp3      exp4        
1    1   0   0   0   1   0  0  0  0  1   7.0  0.4900  0.343000   0.24010000
2    0   0   0   0   1   0  0  0  0  1  31.0  9.6100 29.791000  92.35210000
3    0   0   1   0   0   0  0  0  0  1  18.0  3.2400  5.832000  10.49760000
4    1   0   0   0   0   1  0  0  0  1  25.0  6.2500 15.625000  39.06250000
5    1   0   0   0   1   0  0  0  0  1  22.0  4.8400 10.648000  23.42560000
6    1   0   0   0   1   0  0  0  0  1   1.0  0.0100  0.001000   0.00010000
7    1   0   1   0   0   0  0  0  0  1  42.0 17.6400 74.088000 311.16960000
8    0   0   1   0   0   0  0  0  0  1  37.0 13.6900 50.653000 187.41610000
9    1   0   1   0   0   0  0  0  0  1  31.0  9.6100 29.791000  92.35210000
10   1   0   0   0   1   0  0  0  0  1   4.0  0.1600  0.064000   0.02560000
11   1   0   1   0   0   0  0  0  0  1   7.0  0.4900  0.343000   0.24010000
12   0   0   1   0   0   0  0  0  0  1  30.0  9.0000 27.000000  81.00000000
13   1   0   0   1   0   0  0  0  0  1   5.5  0.3025  0.166375   0.09150625
14   1   0   0   1   0   0  0  0  0  1  20.5  4.2025  8.615125  17.66100625
15   1   0   0   0   1   0  0  0  0  1  25.0  6.2500 15.625000  39.06250000
16   0   0   1   0   0   0  0  0  0  1  16.0  2.5600  4.096000   6.55360000
17   0   0   1   0   0   0  0  0  0  1  27.0  7.2900 19.683000  53.14410000
18   0   0   0   1   0   0  0  0  0  1   3.5  0.1225  0.042875   0.01500625
19   0   0   1   0   0   0  0  0  0  1   6.0  0.3600  0.216000   0.12960000
20   0   0   1   0   0   0  0  0  0  1   8.0  0.6400  0.512000   0.40960000
21   1   0   0   1   0   0  0  0  0  1   6.5  0.4225  0.274625   0.17850625
22   1   0   0   1   0   0  0  0  0  1   8.5  0.7225  0.614125   0.52200625
23   1   0   0   0   0   1  0  0  0  1  11.0  1.2100  1.331000   1.46410000
24   0   0   0   0   1   0  0  0  0  1  11.0  1.2100  1.331000   1.46410000
25   1   0   0   1   0   0  0  0  0  1  24.5  6.0025 14.706125  36.03000625
26   0   0   1   0   0   0  0  0  0  1   8.0  0.6400  0.512000   0.40960000
27   1   0   1   0   0   0  0  0  0  1   5.0  0.2500  0.125000   0.06250000
28   0   0   0   1   0   0  0  0  0  1   4.5  0.2025  0.091125   0.04100625
29   0   0   0   0   1   0  0  0  0  1   8.0  0.6400  0.512000   0.40960000
30   0   0   0   0   1   0  0  0  0  1  26.0  6.7600 17.576000  45.69760000
⋮    ⋮   ⋮   ⋮   ⋮   ⋮   ⋮  ⋮  ⋮  ⋮  ⋮  ⋮    ⋮       ⋮         ⋮           
5121 1   0   0   0   0   1  0  0  1  0   7.0  0.4900  0.343000   0.2401000 
5122 0   0   0   0   0   1  0  0  1  0   2.0  0.0400  0.008000   0.0016000 
5123 1   0   0   1   0   0  0  0  1  0   7.5  0.5625  0.421875   0.3164062 
5124 0   0   0   0   1   0  0  0  1  0   5.0  0.2500  0.125000   0.0625000 
5125 1   0   0   0   0   1  0  0  1  0  24.0  5.7600 13.824000  33.1776000 
5126 0   0   0   1   0   0  0  0  1  0  28.5  8.1225 23.149125  65.9750063 
5127 1   0   1   0   0   0  0  0  1  0  25.0  6.2500 15.625000  39.0625000 
5128 0   0   1   0   0   0  0  0  1  0  19.0  3.6100  6.859000  13.0321000 
5129 0   0   0   0   1   0  0  0  1  0   8.0  0.6400  0.512000   0.4096000 
5130 1   0   1   0   0   0  0  0  1  0  12.0  1.4400  1.728000   2.0736000 
5131 0   0   0   0   0   1  0  0  1  0   6.0  0.3600  0.216000   0.1296000 
5132 1   0   0   0   1   0  0  0  1  0  25.0  6.2500 15.625000  39.0625000 
5133 0   0   1   0   0   0  0  0  1  0  20.0  4.0000  8.000000  16.0000000 
5134 0   0   1   0   0   0  0  0  1  0  17.0  2.8900  4.913000   8.3521000 
5135 0   0   0   1   0   0  0  0  1  0  31.5  9.9225 31.255875  98.4560063 
5136 0   0   1   0   0   0  0  0  1  0  15.0  2.2500  3.375000   5.0625000 
5137 1   0   0   0   1   0  0  0  1  0  34.0 11.5600 39.304000 133.6336000 
5138 1   0   1   0   0   0  0  0  1  0   8.0  0.6400  0.512000   0.4096000 
5139 1   0   0   0   0   1  0  0  1  0   8.0  0.6400  0.512000   0.4096000 
5140 0   0   0   0   1   0  0  0  1  0  28.0  7.8400 21.952000  61.4656000 
5141 1   0   0   0   1   0  0  0  1  0   5.0  0.2500 

7. In some situations it can be useful to generate arrays (matrices) in R. However, for regressions, it is standard to use formulas. We will still be generating collections and matrices just to get used to doing so in R

In [12]:
y <- data$lwage

8.
8.1

In [13]:
X_basic <- model.matrix(~ sex + hsg + scl + clg + ad + so + we + ne + exp1 + C(occ2) + C(ind2), data = data)

8.2

In [14]:
X_flexible <- model.matrix(~ sex + (exp1 + exp2 + exp3 + exp4) * (hsg + scl + clg + ad + so + we + ne + C(occ2) + C(ind2)), data = data)

8.3

In [15]:
X_extra_flexible <- model.matrix(~ sex + (exp1 + exp2 + exp3 + exp4 + hsg + scl + clg + ad + so + we + ne + C(occ2) + C(ind2)) ^ 2, data = data)

9.

In [16]:
train_sample <- runif(length(y)) < 0.8

In [17]:
test_sample <- !train_sample

10.

In [18]:
basic_model <- lm(lwage ~ sex + hsg + scl + clg + ad + so + we + ne + exp1 + C(occ2) + C(ind2), data = data, subset = train_sample)

In [19]:
flexible_model <- lm(lwage ~ sex + (exp1 + exp2 + exp3 + exp4) * (hsg + scl + clg + ad + so + we + ne + C(occ2) + C(ind2)), data = data, subset = train_sample)

In [20]:
extra_flexible_model <- lm(lwage ~ sex + (exp1 + exp2 + exp3 + exp4 + hsg + scl + clg + ad + so + we + ne + C(occ2) + C(ind2)) ^ 2, data = data, subset = train_sample)

11.

In [21]:
mse_train_basic <- mean(basic_model$residuals ^ 2)
r2_train_basic <- 1 - mse_train_basic / var(y[train_sample])
adjr2_train_basic <- 1 - length(y) / (length(y) - ncol(X_basic)) * mse_train_basic / var(y)
mse_test_basic <- mean((predict(basic_model, subset(data, subset = test_sample)) - y[test_sample]) ^ 2)
r2_test_basic <- 1 - mse_test_basic / var(y[test_sample])

print(paste("Training MSE for basic model:", mse_train_basic))
print(paste("Training R2 for basic model:", r2_train_basic))
print(paste("Adjusted training R2 for basic model:", adjr2_train_basic))
print(paste("Testing MSE for basic model:", mse_test_basic))
print(paste("Testing R2 for basic model:", r2_test_basic))

Warning message:
"contrasts dropped from factor C(occ2)"
Warning message:
"contrasts dropped from factor C(ind2)"


[1] "Training MSE for basic model: 0.227201988456661"
[1] "Training R2 for basic model: 0.300890323215889"
[1] "Adjusted training R2 for basic model: 0.294660048385198"
[1] "Testing MSE for basic model: 0.216007148303121"
[1] "Testing R2 for basic model: 0.339786285312131"


In [22]:
mse_train_flexible <- mean(flexible_model$residuals ^ 2)
r2_train_flexible <- 1 - mse_train_flexible / var(y)
adjr2_train_flexible <- 1 - length(y) / (length(y) - ncol(X_flexible)) * mse_train_flexible / var(y)
mse_test_flexible <- mean((predict(flexible_model, subset(data, subset = test_sample)) - y[test_sample]) ^ 2)
r2_test_flexible <- 1 - mse_test_flexible / var(y[test_sample])

print(paste("Training MSE for flexible model:", mse_train_flexible))
print(paste("Training R2 for flexible model:", r2_train_flexible))
print(paste("Adjusted training R2 for flexible model:", adjr2_train_flexible))
print(paste("Testing MSE for flexible model:", mse_test_flexible))
print(paste("Testing R2 for flexible model:", r2_test_flexible))

Warning message:
"contrasts dropped from factor C(occ2)"
Warning message:
"contrasts dropped from factor C(ind2)"
Warning message in predict.lm(flexible_model, subset(data, subset = test_sample)):
"prediction from rank-deficient fit; attr(*, "non-estim") has doubtful cases"


[1] "Training MSE for flexible model: 0.212092321684295"
[1] "Training R2 for flexible model: 0.348087835672244"
[1] "Adjusted training R2 for flexible model: 0.315385879631333"
[1] "Testing MSE for flexible model: 0.234677389836864"
[1] "Testing R2 for flexible model: 0.282721740856339"


In [23]:
mse_train_extra_flexible <- mean(extra_flexible_model$residuals ^ 2)
r2_train_extra_flexible <- 1 - mse_train_extra_flexible / var(y)
adjr2_train_extra_flexible <- 1 - length(y) / (length(y) - ncol(X_extra_flexible)) * mse_train_extra_flexible / var(y)
mse_test_extra_flexible <- mean((predict(extra_flexible_model, subset(data, subset = test_sample)) - y[test_sample]) ^ 2)
r2_test_extra_flexible <- 1 - mse_test_extra_flexible / var(y[test_sample])

print(paste("Training MSE for flexible model:", mse_train_extra_flexible))
print(paste("Training R2 for flexible model:", r2_train_extra_flexible))
print(paste("Adjusted training R2 for flexible model:", adjr2_train_extra_flexible))
print(paste("Testing MSE for flexible model:", mse_test_extra_flexible))
print(paste("Testing R2 for flexible model:", r2_test_extra_flexible))

Warning message:
"contrasts dropped from factor C(occ2)"
Warning message:
"contrasts dropped from factor C(ind2)"
Warning message in predict.lm(extra_flexible_model, subset(data, subset = test_sample)):
"prediction from rank-deficient fit; attr(*, "non-estim") has doubtful cases"


[1] "Training MSE for flexible model: 0.176004618226188"
[1] "Training R2 for flexible model: 0.459011289572721"
[1] "Adjusted training R2 for flexible model: 0.331872455947126"
[1] "Testing MSE for flexible model: 0.297084951605963"
[1] "Testing R2 for flexible model: 0.0919765340247114"
